esp-hal-common/src/parl_io.rs

```diff
@@ -1924,5 +2086,49 @@ mod private {
 
             reg_block.rx_genrl_cfg.read().rx_eof_gen_sel().bit_is_set()
         }
+
+        #[cfg(feature = "async")]
```

Comment:
Any reason to cfg these behind the async flag? They seem useful outside of async too.

Reply:
that was just to mute warnings in the non-async case - was considering to use allow-unused instead 🤔

Reply:
Oh, sorry I thought these are impl on the parlio driver itself, I now see they're private so we can leave them as is I think :)